Library/Homebrew/extend/os/mac/keg.rb

```diff
@@ -31,36 +31,38 @@ def codesign_patched_binary(file)
     return unless Hardware::CPU.arm?
 
     odebug "Codesigning #{file}"
-    # Use quiet_system to squash notifications about resigning binaries
-    # which already have valid signatures.
-    return if quiet_system("codesign", "--sign", "-", "--force",
-                           "--preserve-metadata=entitlements,requirements,flags,runtime",
-                           file)
+    file.ensure_writable do
```

Comment:
I don't think this fixes #13903. The problem in #13903 is not that the file we want to `codesign` is not writeable, but that files next to it that `codesign` needs to modify are not writeable.

In particular, when doing `codesign ... Chisel.framework/Chisel`, what we need to be writeable is actually `Chisel.framework/_CodeSignature/CodeResources` and not `Chisel.framework/Chisel`.

See Homebrew/discussions#3669 for more details.

Reply:
Ah, thanks for the additional context. Are such ancillary files only present when codesigning a framework? From the `chisel` example so far, I take it that codesigning a dylib or other single binary is still self-contained with no ancillary files.